tests.py

```diff
@@ -334,6 +334,32 @@ def test_unregister_unregisters_multiple(self):
         self.db.unregister_model(self.MyModel)
         assert self.db.model_registry == {}
 
+    def test_add_column_doesnt_break_anything(self):
+        self.db.run("ALTER TABLE foo ADD COLUMN boo text")
+        one = self.db.one("SELECT foo.*::foo FROM foo WHERE bar='baz'")
+        assert one.boo is None
+
+    def test_replace_column_different_type(self):
+        self.db.run("CREATE TABLE grok (bar int)")
+        self.db.run("INSERT INTO grok VALUES (0)")
+        class EmptyModel(Model): pass
+        self.db.register_model(EmptyModel, 'grok')
+        # Add a new column then drop the original one
+        self.db.run("ALTER TABLE grok ADD COLUMN biz text NOT NULL DEFAULT 'x'")
+        self.db.run("ALTER TABLE grok DROP COLUMN bar")
+        # The number of columns hasn't changed but the names and types have
+        one = self.db.one("SELECT grok.*::grok FROM grok LIMIT 1")
+        assert one.biz == 'x'
+        assert not hasattr(one, 'bar')
+
+    @mark.xfail(raises=AttributeError)
+    def test_replace_column_same_type(self):
+        self.db.run("ALTER TABLE foo ADD COLUMN biz int NOT NULL DEFAULT 0")
```

Comment:
This test fails because the number of columns is the same (so the length test doesn't detect the change) and the types of the old and new columns are compatible (so there is no type error to catch).